opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -868,6 +947,54 @@ class EclMaterialLawManager
         }
     }
 
+    template <class Container>
+    void readGasWaterEffectiveParameters_(Container& dest,
+                                        const Opm::EclipseState& eclState,
+                                        unsigned satRegionIdx)
+    {
+        if (!hasGas || !hasWater || hasOil)
+            // we don't read anything if either the gas or the water phase is not active or if oil is present
+            return;
+
+        dest[satRegionIdx] = std::make_shared<GasWaterEffectiveTwoPhaseParams>();
+
+        auto& effParams = *dest[satRegionIdx];
+
+        const auto tolcrit = eclState.runspec().saturationFunctionControls()
+            .minimumRelpermMobilityThreshold();
+
+        const auto& tableManager = eclState.getTableManager();
+
+        switch (eclState.runspec().saturationFunctionControls().family()) {
+        case SatFuncControls::KeywordFamily::Family_I:
+        {
+            // No Family I applicable in this case
+            break;
```

Comment:
Does it make sense to throw here?

Reply:
I agree, that does make sense. I included a throw